TestState.cpp

```diff
@@ -35,30 +33,19 @@ void TestState::onEnter(Game &game) {
   
   
   board = Board(game, position, 8, boardWidth, boardHeight);
-
-  for(auto &piece : pieces.playerPieces) {
-    piece.onEnter(game, board);
-  }
-
-  for(auto &piece : pieces.enemyPieces) {
-    piece.onEnter(game, board);
-  }
-
 }
 
 void TestState::onEvent(sf::Event event) {
-  for(auto &piece : pieces.playerPieces) {
-    piece.onEvent(event, board);
-  }
-
+  board.onEvent(event);
 }
 
 std::shared_ptr<State> TestState::update(sf::RenderWindow& window) {
 
   board.update(window);
 
-  for(auto &piece : pieces.playerPieces) {
-    piece.update(window, board);
+  for(auto &piece : board.pieces) {
```

Comment:
now that the enemy pieces and player pieces are combined, I can move both set of the pieces. We will have to be able to know who is the owner of a piece.

Reply:
On the Piece class there is a `player` integer